src/creatures/monsters/monster.cpp

```diff
@@ -1414,39 +1414,62 @@ void Monster::onThinkSound(uint32_t interval) {
 }
 
 bool Monster::pushItem(const std::shared_ptr<Item> &item, const Direction &nextDirection) {
+	if (!item) {
+		return false;
+	}
+
 	const Position &centerPos = item->getPosition();
+	std::shared_ptr<Tile> fromTile = std::dynamic_pointer_cast<Tile>(item->getParent());
```

Comment:
std::dynamic_pointer_cast<Tile>(item->getParent()) is not ideal here.

Performance: dynamic_pointer_cast incurs RTTI cost and refcount adjustments unnecessarily.

Correctness: This only works if the item’s direct parent is a Tile. If the item is inside a container on the ground, getParent() will return the container, and the cast will fail.

Better alternative: Use item->getTile(). It already traverses the parent hierarchy to find the actual Tile containing the item, works for both loose items and items inside containers, and avoids RTTI overhead.

```cpp
std::shared_ptr<Tile> fromTile = item->getTile();
if (!fromTile) {
    return false;
}
```
This is cleaner, faster, and handles more cases correctly.